File: CountPrimes.py
class Solution:
    # @param {integer} n
    # @return {integer}
    def countPrimes(self, n):
        numbers = [0, 0] + [1 for i in range(2, n)]
        for i in range(2,int(n**0.5)):
            if numbers[i]==0:
                continue
            t = 2
            while t * i < n:
                numbers[t*i] = 0
                t += 1
        count = 0
        for i in range(2, n):
            if numbers[i] == 1:
                count += 1
        return count

print(Solution().countPrimes(10000000))

File: cpp/CountPrimes.cpp
#include<iostream>
#include<cmath>
#include<cstring>
using namespace std;
class Solution {
public:
    int countPrimes(int n) {
        bool* numbers = new bool[n];
        memset(numbers, true, n);
        int sqrtn = int(sqrt(n));
        int count = 0;
        for(int i = 2; i <= sqrtn; i++){
            if(numbers[i] == false)
                continue;
            int times = 2;
            while(times * i < n){
                numbers[times * i] = false;
                times++;
            }
        }
        for(int i = 2; i < n; i++){
            if(numbers[i])
                count += 1;
        }
        delete numbers;
        return count;
    }
};

int main(){
    cout<<Solution().countPrimes(1000000)<<endl;
    return 0;
}
